crates/ruff_linter/src/linter.rs

```diff
@@ -445,7 +445,7 @@ pub fn lint_only(
             &locator,
             &directives,
         ),
-        has_syntax_error: !parsed.is_valid(),
+        has_syntax_error: !parsed.is_valid() || !parsed.unsupported_syntax_errors().is_empty(),
```

Comment:
I wonder if we should have a separate version of `is_valid` method that checks both `ParseError` and `UnsupportedSyntaxError` which can then be used here. I'm not sure what to name the method though as both are syntax errors at the end 😅